pkg/resource/deploy/source_eval.go

```diff
@@ -1190,10 +1190,6 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	var outputDeps map[string]*pulumirpc.RegisterResourceResponse_PropertyDependencies
 	if remote {
 		provider, ok := rm.providers.GetProvider(providerRef)
-		if providers.IsDenyDefaultsProvider(providerRef) {
-			msg := diag.GetDefaultProviderDenied(resource.URN(t.String())).Message
-			return nil, fmt.Errorf(msg, t.Package().String(), t.String())
-		}
```

Comment:
This doesn't need to be reverted. It only functions to make an inscrutable error message coherent.